stdlib/public/Atomics/AtomicIntegerConformances.swift.gyb

```diff
@@ -0,0 +1,235 @@
+  ) -> ${swiftType} {
+    switch ordering {
+%   for (swiftOrder, llvmOrder) in loadOrderings():
+    ${caseStatementForOrdering(swiftOrder)}:
+      return ${swiftType}(Builtin.atomicload_${llvmOrder}_${builtinType}(
+          pointer._rawValue))
+%   end
+    }
+  }
+
+  @_semantics("atomics.requires_constant_orderings")
+  @_transparent @_alwaysEmitIntoClient
+  public static func atomicStore(
+    _ desired: ${swiftType},
+    at pointer: UnsafeMutablePointer<${swiftType}>,
+    ordering: AtomicStoreOrdering
+  ) {
+    switch ordering {
+%   for (swiftOrder, llvmOrder) in storeOrderings():
+    ${caseStatementForOrdering(swiftOrder)}:
+      Builtin.atomicstore_${llvmOrder}_${builtinType}(
+        pointer._rawValue,
+        desired.${getter})
+%   end
+    }
+  }
+
+  @_semantics("atomics.requires_constant_orderings")
+  @_transparent @_alwaysEmitIntoClient
+  public static func atomicExchange(
+    _ desired: ${swiftType},
+    at pointer: UnsafeMutablePointer<${swiftType}>,
+    ordering: AtomicUpdateOrdering
+  ) -> ${swiftType} {
+    switch ordering {
+%   for (swiftOrder, llvmOrder, _) in rmwOrderings():
+    ${caseStatementForOrdering(swiftOrder)}:
+      return ${swiftType}(Builtin.atomicrmw_xchg_${llvmOrder}_${builtinType}(
+          pointer._rawValue, desired.${getter}))
+%   end
+    }
+  }
+
+  @_semantics("atomics.requires_constant_orderings")
+  @_transparent @_alwaysEmitIntoClient
+  public static func atomicCompareExchange(
+    expected: ${swiftType},
+    desired: ${swiftType},
+    at pointer: UnsafeMutablePointer<${swiftType}>,
+    ordering: AtomicUpdateOrdering
+  ) -> (exchanged: Bool, original: ${swiftType}) {
+    switch ordering {
+%   for (swiftOrder, llvmOrder, failureOrder) in rmwOrderings():
+    ${caseStatementForOrdering(swiftOrder)}:
+      let (oldValue, won) = Builtin.cmpxchg_${llvmOrder}_${failureOrder}_${builtinType}(
+        pointer._rawValue, expected.${getter}, desired.${getter})
+      return (Bool(_builtinBooleanLiteral: won), ${swiftType}(oldValue))
+%   end
+    }
+  }
+
+  @_semantics("atomics.requires_constant_orderings")
+  @_transparent @_alwaysEmitIntoClient
+  public static func atomicCompareExchange(
+    expected: ${swiftType},
+    desired: ${swiftType},
+    at pointer: UnsafeMutablePointer<${swiftType}>,
+    ordering: AtomicUpdateOrdering,
+    failureOrdering: AtomicLoadOrdering
+  ) -> (exchanged: Bool, original: ${swiftType}) {
+    switch (ordering, failureOrdering) {
+% for (swiftSuccess, llvmSuccess, _) in rmwOrderings():
+%   for (swiftFailure, llvmFailure) in loadOrderings():
+    case (.${swiftSuccess}, .${swiftFailure}):
+      let (oldValue, won) = Builtin.cmpxchg_${llvmOrders(llvmSuccess, llvmFailure)}_${builtinType}(
+        pointer._rawValue, expected.${getter}, desired.${getter})
+      return (Bool(_builtinBooleanLiteral: won), ${swiftType}(oldValue))
+%   end
+% end
+    default:
+      let (oldValue, won) = Builtin.cmpxchg_seqcst_seqcst_${builtinType}(
```

Comment:
Shouldn't the `default:` just contain some kind of unreachable marker?

Reply:
Good point. This is a remnant of a CRISP pattern from a previous design iteration.

Reply:
Good point. (This is a remnant of a CRISP from a previous design iteration.)